debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresChangeEventSourceCoordinator.java

```diff
@@ -85,4 +128,20 @@ private void setSnapshotStartLsn(PostgresSnapshotChangeEventSource snapshotSourc
         snapshotSource.updateOffsetForPreSnapshotCatchUpStreaming(offsetContext);
     }
 
+    @Override
+    public void commitOffset(Map<String, ?> partition, Map<String, ?> offset) {
+        if (waitForSnapshotCompletion) {
+            LOGGER.debug("Checking the offset value for snapshot completion");
+            OffsetState offsetState = new PostgresOffsetContext.Loader((PostgresConnectorConfig) connectorConfig).load(offset).asOffsetState();
+            if(!offsetState.snapshotInEffect()) {
+                LOGGER.debug("Offset conveys that snapshot has completed.");
+                waitForSnapshotCompletion = false;
+            }
+        }
+
+        if (!commitOffsetLock.isLocked() && streamingSource != null && offset != null) {
```

Comment:
Would be good to add a comment why would we not execute this block immediately when `waitForSnapshotCompletion` is set to `false`